src/libos/src/fs/dev_fs/dev_fd.rs

```diff
@@ -0,0 +1,48 @@
+use super::*;
+pub struct DevFd;
+
+// Implement /dev/fd as a symlink to /proc/self/fd
+impl INode for DevFd {
+    fn read_at(&self, offset: usize, buf: &mut [u8]) -> vfs::Result<usize> {
+        let proc_fd = "/proc/self/fd";
+        for (tgt, src) in buf.iter_mut().zip(proc_fd.as_bytes().iter()) {
+            *tgt = *src;
+        }
+        Ok(proc_fd.len())
+    }
+
+    fn write_at(&self, offset: usize, buf: &[u8]) -> vfs::Result<usize> {
+        Err(vfs::FsError::NotFile)
```

Comment:
I think the `NotFile` error here is not correct, because symlinks are also allowed to write the new symlink content. In this case, the symlink cannot be modified, we can return 'FsError::PermError'

Reply:
Done.